src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas2/SubmissionsController.kt

```diff
@@ -120,9 +129,23 @@ class SubmissionsController(
     return ResponseEntity(HttpStatus.CREATED)
   }
 
-  private fun processAuthorisationFor(
+  override fun submissionsApplicationIdNotesPost(applicationId: UUID, body: NewCas2ApplicationNote): ResponseEntity<Cas2ApplicationNote> {
+    val noteResult = applicationNoteService.createApplicationNote(applicationId, body)
+
+    val validationResult = processAuthorisationFor(applicationId, noteResult) as ValidatableActionResult<Cas2ApplicationNote>
+
+    val note = processValidation(validationResult) as Cas2ApplicationNoteEntity
+
+    return ResponseEntity
+      .created(URI.create("/cas2/applications/$applicationId/notes/${note.id}"))
+      .body(
+        applicationNotesTransformer.transformJpaToApi(note),
+      )
+  }
+
+  private fun <EntityType> processAuthorisationFor(
     applicationId: UUID,
-    result: AuthorisableActionResult<ValidatableActionResult<Cas2StatusUpdateEntity>>,
+    result: AuthorisableActionResult<ValidatableActionResult<EntityType>>,
```

Comment:
Presumably this change is required since `processAuthorisationFor` was previously only used when an application status was updated?  Therefore since now we are using it for adding an application note we need to refer to a common `EntityType` superclass?